src/components/common/EffectiveDateTime.vue

```diff
@@ -238,7 +238,7 @@ export default class EffectiveDateTime extends Mixins(DateMixin) {
 
   /** Construct the Date Object for storage */
   private constructDate (): void {
-    if (this.isFutureEffective) {
```

Comment:
This failed when dateText was empty string (eg, when changing from immediate to future-effective) -- dateToValidate on line 243 became null and either line 250 or line 257 caused an exception.